autoemulate/experimental/tuner.py

```diff
@@ -23,7 +23,12 @@ class Tuner(InputTypeMixin):
 
     def __init__(self, x: InputLike, y: InputLike | None, n_iter: int):
         self.n_iter = n_iter
-        self.dataset = self._convert_to_dataset(x, y)
+        # Convert input types, convert to tensors to ensure correct shapes, convert back
```

Comment:
Should we maybe have an issue that says we will revisit the dataflow and all the different conversions we do once we finalise the compare loop?